tests/integrations/raftstore/test_custom_quorum.rs

```diff
@@ -1,31 +1,66 @@
 // Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
 
 use futures::future::Future;
+use std::sync::atomic::AtomicBool;
+use std::sync::{mpsc, Arc};
 use std::time::Duration;
 
 use crate::pd_client::PdClient;
+use crate::tikv_util::HandyRwLock;
+use kvproto::raft_serverpb::RaftMessage;
 use raft::eraftpb::ConfChangeType;
+use raft::eraftpb::MessageType;
 use raftstore::store::QuorumAlgorithm;
 use test_raftstore::*;
 
 // Test QuorumAlgorithm::IntegrationOnHalfFail works as expected.
 #[test]
 fn test_integration_on_half_fail_quorum_fn() {
     let mut cluster = new_node_cluster(0, 5);
+    // Disable hibernate region feature so that it will check quorum after some ticks.
+    cluster.cfg.raft_store.hibernate_regions = false;
     cluster.cfg.raft_store.quorum_algorithm = QuorumAlgorithm::IntegrationOnHalfFail;
     cluster.run();
     cluster.must_put(b"k1", b"v0");
 
-    // After peer 4 and 5 fail, no new leader could be elected.
+    let (tx, rx) = mpsc::sync_channel(1024);
+    let msg_cb = Arc::new(Box::new(move |_: &RaftMessage| drop(tx.send(0)))
+        as Box<dyn Fn(&RaftMessage) + Send + Sync>);
+    let filter = Box::new(
+        RegionPacketFilter::new(1, 1)
+            .direction(Direction::Send)
+            .msg_type(MessageType::MsgRequestPreVote)
+            .set_msg_callback(msg_cb)
+            .when(Arc::new(AtomicBool::new(false))),
```

Comment:
Why drop 'MsgRequestPreVote'?

Reply:
They won't be dropped because `when` is `false`.